src/components/views/dialogs/InviteDialog.tsx

```diff
@@ -751,7 +866,179 @@ export default class InviteDialog extends React.PureComponent<IInviteDialogProps
+        })
+            .then((res) => res.json())
+            .then(results => {
+                if (!results) results = [];
+
+                // create a new object and assign user_id, display_name
+                let newObj = { user_id: '', display_name: '', avatar_url: '' }
+                let role_display_name = '';
+                let role_user_id = '';
+                let role_category = '';
+                let roleIsActive = false;
+                newObj = results.map((value) => {
+                    role_display_name = value["displayName"];
+                    newObj.display_name = role_display_name;
+                    role_user_id = value["simplifiedID"];
+                    newObj["user_id"] = role_user_id;
+                    role_category = value["primaryRoleCategoryID"];
+                    // filled or not filled status
+                    // if activePractitionerSet array is non-empty someone is fulfilling that role
+                    if (value.activePractitionerSet?.length >= 1) {
+                        roleIsActive = true;
+                    } else {
+                        roleIsActive = false;
+                    }
+                });
+
+                // update favorite roles
+                this._updateFavoritesForCurrentUser();
+                // update server result mixin(search result) in state
+                if (results.length > 0) {
+                    this.setState({
+                        serverResultsMixin: [...this.state.serverResultsMixin, {
+                            user: new DirectoryMember({
+                                user_id: role_user_id,
+                                display_name: role_display_name,
+                                avatar_url: '',
+                                favorite: false,
+                                available: roleIsActive,
+                                roleCategoryId: role_category
+                            }),
+                            userId: role_user_id,
+                        }],
+                        recents: [],
+                        suggestions: []
+                    });
+                }
+
+                // if filter by favorite is selected, and user tries to search favorite only
+                if (results.length > 0 && this.state.favoriteFilterIsSelected) {
+                    const serverResults = this.state.serverResultsMixin;
+                    let filteredByFavoriteResults;
+                    for (let fav in this.state.favorites) {
+                        filteredByFavoriteResults = serverResults.filter(m => m.user.name.indexOf(fav) !== -1);
+                    }
+                    this.setState({
+                        serverResultsMixin: filteredByFavoriteResults,
+                        recents: [],
+                        suggestions: []
+                    });
+                }
+            }).catch(e => {
+                console.error("Error searching role directory:");
```

Comment:
I think we need to show something in the browser.

Reply:
Thats a valid point. But, matrix default did not show any error on UI for search on prmpt. But, it can be added if you recommend so, I think it is going to require me to add one more variable which is going to carry error. Do you recommend showing error in search pane ? If there is search error then I beleive no result would be shown but yeah it can be added if you recommend too. I will look into including error text on UI.

Reply:
Displaying error if error occurs in search has been implemented and tested and pushed.

Reply:
What does `errorText: err` end up showing in the UI?  We want to show a nice message to the user and not the raw javascript error.

Reply:
NO worries, implemented that now.

Reply:
Update: I have updated error updates from all catch blocks to plain english format to be consistant with matrix code.